Adds limitation item about anomaly detection jobs and frozen tier data (#2312)
elastic/stack-docs: docs/en/stack/ml/anomaly-detection/ml-limitations.asciidoc

in {dfeeds} or jobs. See
{ref}/searching_a_frozen_index.html[Searching a frozen index].

[discrete]

> [[ml-frozen-tire-limitations]]

Author: ```suggestion
[[ml-frozen-tier-limitations]]
```

> === Frozen tires are not supported

Author: ```suggestion
=== Frozen tiers are not supported
```

> {ref}/data-tiers.html#frozen-tier[Frozen tires] camnot be used in {anomaly-jobs} 

Author: ```suggestion
{ref}/data-tiers.html#frozen-tier[Frozen tiers] cannot be used in {anomaly-jobs} 
```
